centreon/src/Core/Common/Infrastructure/Routing/ModuleRouteLoader.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * Copyright 2005 - 2025 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Common\Infrastructure\Routing;
+
+use Core\Module\Infrastructure\ModuleVersionChecker;
+use Symfony\Bundle\FrameworkBundle\Routing\RouteLoaderInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\Routing\Loader\AttributeFileLoader;
+use Symfony\Component\Routing\RouteCollection;
+
+abstract readonly class ModuleRouteLoader implements RouteLoaderInterface
+{
+    public function __construct(
+        #[Autowire(service: 'routing.loader.attribute.file')] private AttributeFileLoader $loader,
+        #[Autowire(param: 'kernel.project_dir')] private string $projectDir,
+        private ModuleVersionChecker $versionChecker
+    ) {
+    }
+
+    abstract protected function getModuleName(): string;
+
+    abstract protected function getModuleDirectory(): string;
+
+    final public function __invoke(): RouteCollection
+    {
+        if($this->versionChecker->hasANewVersionAvailable($this->getModuleName())) {
```

Comment:
Naming question:

What are we checking ? 
- If Centreon released a new version not yet installed for the module ?
- If the current module installation process has been been done until the end (with SQL migrations & so on)

In the first case the naming is perfect otherwise I would suggest a naming like :

Class name :
`ModuleInstallationVerifier`

Method :
`isFullyInstalled()`

Reply:
Indeed, while we develop it we thought we'll see during the review how to name this class !